linera-core/src/client.rs

```diff
@@ -227,6 +228,10 @@ pub struct ChainState {
     /// This contains blobs belonging to our `pending_block` that may not even have
     /// been processed by (i.e. been proposed to) our own local chain manager yet.
     pub pending_blobs: BTreeMap<BlobId, HashedBlob>,
+
+    /// A mutex that is held whilst we are preparing the next block, to ensure that no
+    /// other client can begin preparing a block.
+    preparing_block: Arc<Mutex<()>>,
```

Comment:
I wonder if this is an indication that we're missing a `BlockPreparer` object with its own (synchronized) state.